internal/handlers/common/projection.go

```diff
@@ -14,10 +14,47 @@
 
 package common
 
-import "github.com/FerretDB/FerretDB/internal/types"
+import (
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
 
 // ProjectDocuments modifies given documents in places according to the given projection.
 func ProjectDocuments(docs []*types.Document, projection *types.Document) error {
-	// TODO
+	if projection.Len() == 0 {
+		return nil
+	}
+	projectionMap := projection.Map()
+	for i := 0; i < len(docs); i++ {
+		for k := range docs[i].Map() {
+			if k == "_id" {
+				continue
+			}
+
+			v, ok := projectionMap[k]
+			if !ok {
+				docs[i].Remove(k)
+				continue
+			}
+
+			switch v := v.(type) {
+			case bool:
+				if !v {
+					docs[i].Remove(k)
+					continue
+				}
+				continue
+
+			case int32, int64, float32, float64:
+				if compareScalars(v, int32(0)) == equal {
```

Comment:
Can we use compareNumbers?

Reply:
then case with int32 , int64 is not handled. `compareScalars` is more convenient here.